tee-worker/litentry/core/assertion-build/src/a14.rs

```diff
@@ -115,24 +116,30 @@ pub fn build(
 	let mut value = false;
 	let mut client = A14Client::new(data_provider_config);
 
-	for address in polkadot_addresses {
-		let data = A14Data {
-			params: A14DataParams { address },
-			include_metadata: false,
-			include_widgets: false,
-		};
-		let response = client.send_request(&data)?;
-
-		let result = response
-			.data
-			.get("result")
-			.and_then(|r| r.as_bool())
-			.ok_or(Error::RequestVCFailed(Assertion::A14, ErrorDetail::ParseError))?;
-		if result {
-			value = result;
-			break
-		}
-	}
+	loop_with_abort_strategy(
+		polkadot_addresses,
+		|address| {
+			let data = A14Data {
+				params: A14DataParams { address: address.clone() },
+				include_metadata: false,
+				include_widgets: false,
+			};
+			let response = client.send_request(&data)?;
+
+			let result = response
+				.data
+				.get("result")
+				.and_then(|r| r.as_bool())
```

Comment:
Do I understand correctly that:

- old version stops if `RequestVCFailed` occours
- new version doesn't stop if `RequestVCFailed` occours 

I checked `test_continue_until_end_success` and that's my understanding. Am I right ?

Reply:
Yes, well spot.
It's changed or fixed because the VC result should be true if any address has result true return from api, and it shouldn't fail if only one of them failed.

Reply:
So if I have 10 addresses and for one of them the result would be true, but due to the network problems request failed I will get VC with `value: [false]`. 

Maybe there should be an additional information saying that not all addresses were checked ?

Honestly, I'm not sure which version is better (previous or current) but it seems like the previous was more accurate.

Reply:
Just wanted to confirm that the behaviour has slightly changed.